java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/verifier/VerifyBuiltArtifactsCommand.java

```diff
@@ -0,0 +1,146 @@
+import static java.nio.file.FileVisitResult.CONTINUE;
+import static org.apache.maven.repository.RepositorySystem.DEFAULT_REMOTE_REPO_ID;
+import static org.apache.maven.repository.RepositorySystem.DEFAULT_REMOTE_REPO_URL;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+
+import org.apache.maven.repository.internal.MavenRepositorySystemUtils;
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.repository.RemoteRepository.Builder;
+
+import com.redhat.hacbs.container.verifier.asm.JarInfo;
+
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Option;
+
+@Command(name = "verify-built-artifacts")
+public class VerifyBuiltArtifactsCommand implements Callable<Integer> {
+    @Option(required = true, names = { "-r", "--repository-url" })
+    String repositoryUrl;
+
+    @Option(required = true, names = { "-d", "--deploy-path" })
+    Path deployPath;
+
+    @Option(names = { "-gs", "--global-settings" })
+    Path globalSettingsFile;
+
+    @Option(names = { "-s", "--settings" })
+    Path settingsFile;
+
+    private final List<RemoteRepository> remoteRepositories;
+
+    private final RepositorySystem system;
+
+    private final DefaultRepositorySystemSession session;
+
+    private boolean failed;
+
+    public VerifyBuiltArtifactsCommand() throws IOException {
+        session = MavenRepositorySystemUtils.newSession();
+        var settings = newSettings(globalSettingsFile, settingsFile);
+        var mirrorSelector = newMirrorSelector(settings);
+        session.setMirrorSelector(mirrorSelector);
+        var authenticationSelector = newAuthenticationSelector(settings);
+        session.setAuthenticationSelector(authenticationSelector);
+        system = newRepositorySystem();
+        var localRepositoryDirectory = Files.createTempDirectory("hacbs-verify-built-artifacts-");
+        var localRepository = new LocalRepository(localRepositoryDirectory.toFile());
+        var manager = system.newLocalRepositoryManager(session, localRepository);
+        session.setLocalRepositoryManager(manager);
+        var centralRepository = new Builder(DEFAULT_REMOTE_REPO_ID, "default", DEFAULT_REMOTE_REPO_URL).build();
+        var mirror = mirrorSelector.getMirror(centralRepository);
+
+        if (mirror != null) {
+            System.out.printf("Mirror for %s: %s%n", centralRepository.getId(), mirror);
+            remoteRepositories = List.of(mirror);
+        } else {
+            System.out.printf("Using repository URL %s%n", repositoryUrl);
```

Comment:
You can't do this in the constructor, the fields will not have been set yet.

Reply:
Fixed it. Thanks.